src/include/miopen/sqlite_db.hpp

```diff
@@ -155,22 +155,60 @@ struct SQLiteSerializable
     }
 };
 
+class SQLite
+{
+    class impl;
+    // do we need propagate const
+    std::unique_ptr<impl> pImpl;
+
+    public:
+    class Statement
+    {
+        class impl;
+        std::unique_ptr<impl> pImpl;
+
+        public:
+        Statement(const SQLite& sql, const std::string& query);
+        Statement(const SQLite& sql,
+                  const std::string& query,
+                  const std::vector<std::string>& vals);
+        Statement();
+        ~Statement();
+        Statement(Statement&&);
+        Statement(const Statement&) = delete;
+        Statement& operator         =(Statement&&);
+        Statement& operator=(const Statement&) = delete;
+        int Step(const SQLite& sql);
+        std::string ColumnText(int idx);
+        std::string ColumnBlob(int idx);
+        int64_t ColumnInt64(int idx);
+        int BindText(int idx, const std::string& txt);
+        int BindBlob(int idx, const std::string& blob);
+        int BindInt64(int idx, int64_t);
+    };
+
+    using SQLRes_t = std::vector<std::unordered_map<std::string, std::string>>;
+    SQLite();
+    SQLite(const std::string& filename_, bool is_system);
+    ~SQLite();
+    SQLite(SQLite&&);
+    SQLite(const SQLite&) = delete;
```

Comment:
Using `unique_ptr` should will automatically delete the constructors necessary so this should no be needed.

Reply:
Fixed